pkg/settings/settings.go

```diff
@@ -0,0 +1,240 @@
+package settings
+
+import (
+	"context"
+	"encoding"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strconv"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/config"
+)
+
+// Getter fetches scoped settings by key.
+type Getter interface {
+	// GetScoped returns the value for key, including checking for any Scope based overrides up to scope.
+	// The tenant for each scope is based on the contexts.CRE in the ctx.
+	GetScoped(ctx context.Context, scope Scope, key string) (value string, err error)
+}
+
+// Registry extends Getter with subscriptions.
+type Registry interface {
+	Getter
+	// SubscribeScoped returns a channel for updates as an efficient alternative to polling.
+	// Stop must be closed to terminate the subscription.
+	SubscribeScoped(ctx context.Context, scope Scope, key string) (updates <-chan Update[string], stop func())
+}
+
+// Setting holds a key, default value, and parsing function for a particular setting.
+// Use Setting.GetOrDefault with a Getter to look up settings.
+// Use Setting.Subscribe with a Registry to have updates pushed over a channel.
+type Setting[T any] struct {
+	Key          string
+	DefaultValue T
+	Scope        Scope
+	Parse        func(string) (T, error) `json:"-" toml:"-"`
```

Comment:
Using plain string values below this layer is simpler and more portable. However, depending on the file format, it can be difficult to get access to the raw string (e.g. TOML numbers), so currently all fields in the files are strings to avoid the issue. If we choose JSON, we should be able to use `json.Number` to support parsing the raw value. On the other hand, many types are not just simple numbers but instead formatted as strings with suffixes, so perhaps the quoted numbers in the TOML are acceptable.